src/membership.md

```diff
@@ -0,0 +1,35 @@
+# Membership
+
+## Library Contributors
+
+Membership to Library Contributors can be offered by the Library Team once
+a regular contributor has made a number of significant contributions over some period of time,
+and has shown to have a good judgement on what changes are acceptable.
+
+## The Library Team and Library API Team
+
+The Library Team and Library API Team pick their own members,
+although it's expected that new members come from the Library Contributors or another Rust team,
+and have already been involved in relevant library work.
+
+## The process
+
+In all cases, the process of adding a new members goes as follows:
+
+1. A member of the Library (API) Team proposes the addition of a contributor on our private mailing list.
+   This includes proposal includes:
+   - A short description of what this person has been working on; how they have been contributing.
+   - A few specific examples of cases where this person clearly communicated their ideas.
+   - A few specific examples that show this person understands what are and what aren't acceptable changes.\
+     Someone who makes significant contributions but usually needs to make large adjustments to their PRs might be a wonderful external contributor,
+     but might not yet be a good match for membership with review persmissions expecting to judge other contributions.
+2. Every single team member is asked for their input. No team member must have any objections.
+   - Objections are ideally shared with the entire team, but may also be shared privately with the team lead or the moderation team.
+   - Objections ideally include examples showing behavior not in line with the expectations described under step 1
+     (or the code of conduct).
+3. The team lead reaches out to the moderation team to ask if they are aware of any objections.
```

Comment:
@rust-lang/mods How do you feel about this? This is not something we did in the past, but I think it'd be good if we do.

Reply:
I _personally_ really like the underlying idea: making it clear to existing members they are encouraged to report issues, rather than have them sulk, be curt, or quit.

In general we'd appreciate if people reported inter-personal issues and let us help them solve them -- at any time, really. Having an explicit reminder they are encouraged to from time to time is an excellent way of "normalizing" this, and having this on the check-list before "forcing" confrontations between individuals is a good "last resort" check, I think.

I'll check with the other mods, but I don't foresee any big issue, and have no wording tweak to suggest at present.